data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -99,8 +100,7 @@ public SqsWorker(final AcknowledgementSetManager acknowledgementSetManager,
 
     @Override
     public void run() {
-
-        while (!Thread.currentThread().isInterrupted()) {
+        while (!isStopped) {
```

Comment:
will this happen for s3 scan shutdown too?

Reply:
No, I checked. Looks like that should work correctly.

Reply:
I missed one `InterruptedException` in s3 scan, will push a new revision for that.